custom-stats-logger/src/main/java/dev/gemfire/stat/logger/StatLogger.java

```diff
@@ -0,0 +1,151 @@
+// Copyright 2023-2024 Broadcom. All rights reserved.
+
+package dev.gemfire.stat.logger;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.geode.StatisticDescriptor;
+import org.apache.geode.Statistics;
+import org.apache.geode.StatisticsType;
+import org.apache.geode.cache.*;
+import org.apache.geode.cache.execute.Function;
+import org.apache.geode.cache.execute.FunctionContext;
+import org.apache.geode.distributed.DistributedSystem;
+import org.apache.geode.logging.internal.log4j.api.LogService;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class StatLogger implements Function,Declarable {
+  private static final Logger logger = LogService.getLogger();
+  private List<StatsHolder> listOfStats = new CopyOnWriteArrayList<>();
```

Comment:
I don't see any need for a CopyOnWriteArrayList, if you change the code to just construct this list during execute and then give that list to the task. That way no other thread can get ahold of it and modify it. It looks like you clear it each time execute is called and that would no longer be needed. Each execute call would just create a new list.

Reply:
Done!